chat/models.py

```diff
@@ -66,38 +66,95 @@ def __str__(self):
+
+        metas = self.puzzle.metas.order_by("created_on").all()
+        if len(metas) > 0:
+            return metas[0].name  # default to the oldest created meta
+        else:
+            None
+
+    def _get_text_category_name(self):
+        return (
+            self._get_category_name()
+            or self.puzzle.hunt.settings.discord_unassigned_text_category
+        )
+
+    def _get_audio_category_name(self):
+        return (
+            self._get_category_name()
+            or self.puzzle.hunt.settings.discord_unassigned_voice_category
+        )
+
     def create_channels(self):
         service = self.get_service()
         update_fields = []
         if self.text_channel_id is None:
-            self.text_channel_id = service.create_text_channel(self.name)
+            self.text_channel_id = service.create_text_channel(
+                self.get_guild_id(), self.name, self._get_text_category_name()
+            )
             self.text_channel_url = service.create_channel_url(
-                self.text_channel_id, is_audio=False
+                self.get_guild_id(), self.text_channel_id, is_audio=False
             )
             update_fields.extend(["text_channel_id", "text_channel_url"])
 
         if self.audio_channel_id is None:
-            self.audio_channel_id = service.create_audio_channel(self.name)
+            self.audio_channel_id = service.create_audio_channel(
+                self.get_guild_id(), self.name, self._get_audio_category_name()
+            )
             self.audio_channel_url = service.create_channel_url(
-                self.audio_channel_id, is_audio=True
+                self.get_guild_id(), self.audio_channel_id, is_audio=True
             )
             update_fields.extend(["audio_channel_id", "audio_channel_url"])
 
         self.save(update_fields=update_fields)
 
     def archive_channels(self):
+        service = self.get_service()
+        archive_category = self.puzzle.hunt.settings.discord_archive_category
+        if self.text_channel_id:
+            service.archive_channel(
+                self.get_guild_id(), self.text_channel_id, archive_category
+            )
+        if self.audio_channel_id:
+            service.archive_channel(
+                self.get_guild_id(), self.audio_channel_id, archive_category
+            )
+
+    def update_category(self):
         service = self.get_service()
         if self.text_channel_id:
-            service.archive_channel(self.text_channel_id)
+            service.categorize_channel(
+                self.get_guild_id(),
+                self.text_channel_id,
+                self._get_text_category_name(),
+            )
         if self.audio_channel_id:
-            service.archive_channel(self.audio_channel_id)
+            service.categorize_channel(
+                self.get_guild_id(),
+                self.audio_channel_id,
+                self._get_audio_category_name(),
+            )
 
     def unarchive_channels(self):
```

Comment:
Ugh yeah, this mess is because the generic ChatService interface originally only had `archive_channel` and `unarchive_channel` which did different things when we were using Slack. Then I added `categorize_channel` for Discord and actually for Discord, all three of those things are actually the same thing =|

I went back and forth a few times about whether I just want to forget about Slack and have a single `categorize_channel` or keep the three separate. What do you think?